backend/kale/common/podutils.py

```diff
@@ -300,3 +307,63 @@ def compute_component_id(pod):
     component_id = component_name + "@sha256=" + component_spec_digest
     log.info("Computed component ID: %s", component_id)
     return component_id
+
+
+def get_snapshotclasses(label_selector=""):
+    """List snapshotclasses."""
+    co_client = k8sutils.get_co_client()
+
+    snapshotclasses = co_client.list_cluster_custom_object(
+        group="snapshot.storage.k8s.io",
+        version="v1beta1",
+        plural="volumesnapshotclasses",
+        label_selector=label_selector)
+    return snapshotclasses
+
+
+def list_snapshotclass_storage_provisioners(label_selector=""):
+    """List the storage provisioners of the snapshotclasses."""
+    snapshotclasses = get_snapshotclasses(label_selector)["items"]
+    snapshotclass_provisioners = []
+    for i in snapshotclasses:
+        snapshotclass_provisioners.append(i["driver"])
+    return snapshotclass_provisioners
+
+
+def check_snapshot_availability():
+    """Check if snapshotclasses are available for notebook."""
+    client = k8sutils.get_v1_client()
+    namespace = get_namespace()
+    pod_name = get_pod_name()
+    pod = client.read_namespaced_pod(pod_name, namespace)
+    snapshotclass_provisioners = list_snapshotclass_storage_provisioners()
+
+    for volume in pod.spec.volumes:
+        pvc = volume.persistent_volume_claim
+        if not pvc:
+            continue
+        pvc = client.read_namespaced_persistent_volume_claim(pvc.claim_name,
+                                                             namespace)
+        ann = pvc.metadata.annotations
+        provisioner = ann.get("volume.beta.kubernetes.io/storage-provisioner",
+                              None)
```

Comment:
No need to `ann.get("..", None)`. `get` already returns `None` by default, so you can just `ann.get("...")`

Reply:
Same here, this is how it is done in Kale currently. 
https://github.com/kubeflow-kale/kale/blob/07ccdc5bdc4c39a3c04dad44d4c07447378efb98/backend/kale/common/podutils.py#L168

Reply:
So we did it wrong the first time :)